### PR TITLE
ULP-1810/ULP-1880: new SMS hook params

### PR DESCRIPTION
**Pull request by sebadoom**
### Description
Update SMS hook parameters to what was agreed with Product.

### References
- Jira: https://auth0team.atlassian.net/browse/ULP-1880
- Jira parent: https://auth0team.atlassian.net/browse/ULP-1810
- Related: https://github.com/auth0/auth0-hooks-templates/pull/7

### Testing

- [x] This change adds test coverage for new/changed/fixed functionality

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used, if not `master`

### CODE REVIEW

**Comment by aaguiarz** [comment]
@sebadoom is this updating the 'sample' hook that is created when creating a new one?

**Comment by sebadoom** [comment]
> @sebadoom is this updating the 'sample' hook that is created when creating a new one?

No, that is here https://github.com/auth0/auth0-hooks-templates/pull/7 which is linked in the same Jira ticket.

File: lib/compilers/send-phone-message.js [hunk 1]
```diff
@@ -45,17 +41,20 @@ function sendPhoneMessageHandler (func, webtaskContext, cb) {
...
+            if (typeof webtaskContext.body.context.client !== 'object') {
+                return cb(new Error('Body.context.client received by extensibility point is not an object'));
+            }
+            if (typeof webtaskContext.body.context.client.client_id !== 'string') {
+                return cb(new Error('Body.context.client.client_id received by extensibility point is not a string'));
+            }
+            if (typeof webtaskContext.body.context.client.name !== 'string') {
+                return cb(new Error('Body.context.client.name received by extensibility point is not a string'));
+            }
+            if (typeof webtaskContext.body.context.client.client_metadata !== 'object') {
```
Comment (joseluisdiaz) [inline lib/compilers/send-phone-message.js]: Maybe make this optional?

File: lib/compilers/send-phone-message.js [hunk 2]
```diff
@@ -21,17 +21,13 @@ function sendPhoneMessageHandler (func, webtaskContext, cb) {
...
-            return cb(new Error('Body.context.factor_type received by extensibility point is not `first` or `second`'));
-        }
         if (!((webtaskContext.body.context.message_type === 'sms') ||
             webtaskContext.body.context.message_type === 'voice')) {
             return cb(new Error('Body.context.message_type received by extensibility point is not `sms` or `voice`'));
         }
         if (!((webtaskContext.body.context.action === 'enrollment') ||
-            webtaskContext.body.context.action ==='authentication')) {
-            return cb(new Error('Body.context.action received by extensibility point is not `enrollment` or `authentication`'));
+            webtaskContext.body.context.action ==='second-factor-authentication')) {
```
Comment (joseluisdiaz) [inline lib/compilers/send-phone-message.js]: Maybe we should only check for string? to have some flexibility here?

Comment (sebadoom) [reply lib/compilers/send-phone-message.js]: Why? We will still need to update a bunch of places if we change this.

Comment (joseluisdiaz) [reply lib/compilers/send-phone-message.js]: Yes. I'm fine with this.

**Comment by lzychowski** [comment]
@sebadoom could you please bump version in `package.json`?

**Comment by sebadoom** [comment]
> @sebadoom could you please bump version in `package.json`?

Done :)

**Comment by sebadoom** [comment]
> @sebadoom could you please bump version in `package.json`?
@lzychowski Done :)

**Pull request merged**